content/gsoc/2024.md

```diff
@@ -0,0 +1,606 @@
+
+### Skills
+The participant should be comfortable with the C++ for Cutter and C for Rizin. They should also be familiar with Qt framework. Experience in GUI code architecture, for example using functional reactive programming or Elm-like approaches is a plus.
+
+### Difficulty
+Advanced
+
+### Benefits for the participant
+The participant will gain an experience of creating complex event-driven software in both C and C++ languages.
+
+### Benefits for the project
+It will allow to work on big files effortlessly in Cutter, will improve analysis quality as well.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: Implement events everywhere in the relevant places across Rizin code and event-driven
+	interaction with Cutter.
+- Final term: Add support for the Cutter interface refresh based on the events from Rizin, implement analysis in background.
+
+### Mentors
+- thestr4ng3r
+
+## Heap viewer completion (175 hour project)
+
+Thanks to the work that was done in the previous GSoC, Cutter and Rizin have nice visualizations of the heap and memory maps. We would like to expand on this feature with performance improvements to the heap parsers and support more memory allocators.
+
+### Task
+ - Complete Cutter's implementation of the windows heap widget #[2723](https://github.com/rizinorg/cutter/pull/2723)
+ - Improve the performance of the Windows heap parser
+ - Fix Windows heap parsing errors
+ - Make the implementation work with remote debugging modes
+
+### Skills
+The participant should be comfortable with the C++, and be familiar with Qt framework
+
+### Difficulty
+Medium
+
+### Benefits for the participant
+The participant will gain the understanding on how modern runtimes provide the heap for various
+programs, which will be beneficial for the binary exploitation skills.
+
+### Benefits for the project
+It will greatly improve the debugging and reverse engineering experience for complex programs,
+also provides the way to design the exploitation techniques with the help of Rizin/Cutter.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: Design and implement heap visualization widgets, add Rizin test and fixes
+- Final term: Various bugfixes related to the heap inspection support on various platforms and allocators, tests and documentation.
+
+### Mentors
+- xvilka
+- Megabeets
+
+### Links/Resources
+- [Issue #1041](https://github.com/rizinorg/cutter/issues/1041)
+- [Heap Viewer plugin for IDA Pro](https://github.com/danigargu/heap-viewer)
+- [Heap parsing for MacOS, tmalloc, jmalloc](https://github.com/rizinorg/rizin/issues/208)
+- [Dynamic Allocator Detection](https://github.com/rizinorg/rizin/issues/157)
+- ["heap"-marked Rizin issues](https://github.com/rizinorg/rizin/labels/heap)
+
+## Diffing mode (175 hour project)
```

Comment:
this is going to be implemented by me. we can remove it

Reply:
Lets keep this for now, we can remove if/when we are accepted.

Reply:
ok